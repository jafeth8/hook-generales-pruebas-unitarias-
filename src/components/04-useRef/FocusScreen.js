import React, { useRef } from 'react'
import "../02-useEffect/Effects.css"
export const FocusScreen = () => {

    const refForm = useRef();

    console.log(refForm);

    const handleClick=()=> {
        //document.querySelector('input').select();
        refForm.current.select();
        console.log(refForm)
    }

    return (
        <div>
            <h1>UseRef</h1>
            <hr/>
            <input type="text" placeholder="tu nombre" className="form-control" ref={refForm}/>

            <button className="btn btn-primary mt-5" onClick={handleClick}>seleccionar input</button>
        </div>
    )
}
