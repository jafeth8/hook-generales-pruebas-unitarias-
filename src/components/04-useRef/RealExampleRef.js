import React, { useState } from 'react'
import "../02-useEffect/Effects.css"
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'

export const RealExampleRef = () => {

    const [show, setShow] = useState(true)

    const handleClick=()=> {
       setShow(!show);
    }



    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr/>
            
            {show?<MultipleCustomHooks/>:null}

            <button className="btn btn-danger mt-5" onClick={handleClick}>mostrar/ocultar</button>
        </div>
    )
}
