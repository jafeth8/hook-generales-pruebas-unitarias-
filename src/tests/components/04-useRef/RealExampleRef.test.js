import {shallow} from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('pruebas en el componente <RealExampleRef/>', () => {

    const wrapper=shallow(<RealExampleRef/>)
    
    test('mostrar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
    
    test('ocultar y mostrar el componente <MultipleCustomHooks/> al presionar el button', () => {
        
        wrapper.find('button').simulate('click')
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
        wrapper.find('button').simulate('click')
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
    
})
