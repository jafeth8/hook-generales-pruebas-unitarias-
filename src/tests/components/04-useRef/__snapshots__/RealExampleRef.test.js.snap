// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <RealExampleRef/> mostrar correctamente el componente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-danger mt-5"
    onClick={[Function]}
  >
    mostrar/ocultar
  </button>
</div>
`;
